src/server/node_manager.c

```diff
@@ -1282,6 +1282,17 @@ set_vnode_state(struct pbsnode *pnode, unsigned long state_bits, enum vnode_stat
 		goto fn_fire_event;
 	}
 
+	unsigned long bits;
+	if (type == Nd_State_And)
+		bits = ~state_bits;
+	else
+		bits = state_bits;
```

Comment:
When changing the vnode state using `pbsnodes -(o|-r)` or `qmgr set node <node> state [+|-]=offline` , the operation type is always `ND_State_Set`.  A condition needs to be added for `ND_State_Set` which sets `bits` to the exclusive-or of the previous and new states so that bits being cleared during a set operation are detected.